astropy/uncertainty/core.py

```diff
@@ -6,9 +6,21 @@
 import builtins
 
 import numpy as np
+from numpy.lib.stride_tricks import DummyArray
 
 from astropy import stats
 from astropy import units as u
+from astropy.utils.compat.numpycompat import NUMPY_LT_1_23, NUMPY_LT_2_0
+
+if NUMPY_LT_2_0:
+    from numpy.core.multiarray import normalize_axis_index
+    from numpy.lib.function_base import _parse_gufunc_signature
+else:
+    from numpy.lib._function_base_impl import _parse_gufunc_signature
+    from numpy.lib.array_utils import normalize_axis_index
```

Comment:
I don't think the CI is currently running against numpy-dev, is it?  Should we consider adding that in this PR since it would then exercise these details, or is it better to just leave this be since the timescale for numpy 2.0 is uncertain?

Reply:
We do pull numpy dev wheel in the devdeps job. Numpy 2.0 is scheduled to be out around Dec 2023.

Reply:
We have been making similar changes for numpy-dev in other packages, just ensuring the -dev run passes!